src/volatile_memory.rs

```diff
@@ -955,22 +956,20 @@ where
         &self.bitmap
     }
 
-    /// Does a volatile write of the value `v` to the address of this ref.
     #[inline(always)]
-    pub fn store(&self, v: T) {
+    unsafe fn write_volatile(&self, addr: *mut Packed<T>, v: T) {
         // SAFETY: Safe because we checked the address and size when creating this VolatileRef.
-        unsafe { write_volatile(self.addr, Packed::<T>(v)) };
+        unsafe { write_volatile(addr, Packed::<T>(v)) };
```

Comment:
The function is marked as `unsafe fn`, so is `unsafe` within the function body redundant?

Reply:
Found same issue at few other places, fixed all of those too. Thanks.

Reply:
It is (for now!) But that's gonna change in the future: https://github.com/rust-lang/rust/issues/71668

In any case, now it's only a lint thus not wrong or invalid